crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -365,7 +374,11 @@ async fn get_vectors_with_chunker(
         values.append_slice(&inner); // I believe this is a clone under the hood.
 
         // Get the length of the last chunk
-        let last_chunk_length = embedded_data.as_slice()[curr + chunks_in_row - 1].len();
+        let last_chunk_length = embedded_data
+            .as_slice()
+            .get(curr + chunks_in_row - 1)
```

Comment:
There's no chance that a chunk would be empty, right? Not sure if this would screw up the math

Reply:
This looks very dangerous :)

Reply:
The last chunk could be 0. The reason we calculate it is we cannot infer the last index from the starting offsets. Take this example
```rust
assert_eq!(chunk_offsets_to_col_values(
    &[0, 4, 7, 12], 3),
    [[0, 4], [4, 7], [7, 12], [12, 15]]
)
```
Without the last length `3`, we cannot know what the second value in last offset pair `[12, 15]`  should be.